internal/handlers/todo_test.go

```diff
@@ -0,0 +1,87 @@
+package handlers_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/shreyas1307/go-server/internal/handlers"
+	"github.com/shreyas1307/go-server/internal/store/models"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockServerStore struct{}
+
+func (m *MockServerStore) AddTodo(ctx context.Context, todo models.Todo) error {
+	return nil
+}
+
+func (m *MockServerStore) GetTodos(ctx context.Context) (models.TodoList, error) {
+	return models.TodoList{}, nil
+}
+
+func TestAddTodoSuccessful(t *testing.T) {
+	th := &handlers.TodoHandler{
```

Comment:
I'll teach you another thing soon (but google it first) - table tests. I'm not a fan of them but they're worth knowing and seeing if you prefer it.